driver-async/src/main/com/mongodb/async/client/gridfs/GridFSBucket.java

```diff
@@ -136,6 +136,37 @@
     GridFSUploadStream openUploadStream(String filename, GridFSUploadOptions options);
 
     /**
+     * Opens a AsyncOutputStream that the application can write the contents of the file to.
+     * <p>
+     * As the application writes the contents to the returned Stream, the contents are uploaded as chunks in the chunks collection. When
+     * the application signals it is done writing the contents of the file by calling close on the returned Stream, a files collection
+     * document is created in the files collection.
+     * </p>
+     *
+     * @param id the custom id value of the file
+     * @param filename the filename for the stream
+     * @return the GridFSUploadStream that provides the ObjectId for the file to be uploaded and the Stream to which the
+     * application will write the contents.
+     */
+    GridFSUploadStream openUploadStreamWithId(BsonValue id, String filename);
```

Comment:
We haven't done that for other methods eg: `openDownloadStreamByName`.
Also, having the same method name but one doesn't return any value seems odd.  Given that want me to rename?